### DISCUSSION TITLE
AttributeError: 'super' object has no attribute '_LinearRegressionModel__init'

### DISCUSSION THREAD

**Discussion by MaryamSaremiKermanshahi** [mrdbourke/pytorch-deep-learning#1088] [Q&A]
Hey, I am trying 01_pytorch_workflow on my colab, 
when I want to check my LinearRegressionModel, with rhis code:

from torch import nn
#Creating linear regression model
class LinearRegressionModel(nn.Module): #almost everything in PyTorch inherits from nn.Module
  def __init__(self):
    super().__init()

    #initiallizing model parameters
    self.weight = nn.Parameter(torch.randn(1,
                                          requires_grad=True,
                                          dtype=torch.float))

    self.bias = nn.Parameter(torch.randn(1,
                                        require_grad=True,
                                        dtype=torch.float))
    #Forward method to define the computation in the model
    def forward(self, x: torch.Tensor) -> torch.Tensor: #<- "x" is the input data
      return self.weights * x + self.bias #this is the linear regression formula

#Create a random seed
torch.manual_seed(42)

#Create an instance of the model (this is a subclass of nn.Module)
model_0 = LinearRegressionModel()

model_0

I face to this error:
AttributeError: 'super' object has no attribute '_LinearRegressionModel__init'

"Would you please help me, get rid of the error"

**Comment by LuluW8071** [comment]
@MaryamSaremiKermanshahi I think you didnt put double underscore on right side of `super(),__init__` causing `AttributeError: 'super' object has no attribute`

Here's the corrected version:

```py
# Create a linear regression model class
class LinearRegressionModel(nn.Module):         # nn.Module contains building blocks(subclass) of neural networks
  def __init__(self):
    super().__init__()
    self.weights = nn.Parameter(torch.randn(1,
                                           requires_grad = True,   # set parameter to be updated via gradient descent -> True
                                           dtype = torch.float))
    self.bias = nn.Parameter(torch.randn(1,
                                         requires_grad = True,     # set parameter to be updated via gradient descent -> True
                                         dtype = torch.float)) 

  # Forward method to define the computation in model
  def forward(self, x:torch.Tensor) -> torch.Tensor:    # 'x' is input data
    return self.weights * x + self.bias   # Linear regression formula
```

**Reply by MaryamSaremiKermanshahi** [reply]
Thank you
